src/agent/service/nodesHolePunchMessageSend.ts

```diff
@@ -32,7 +32,7 @@ function nodesHolePunchMessageSend({
         // If so, ask the nodeManager to relay to the node
       } else if (
         await nodeManager.knowsNode(
-          nodesUtils.makeNodeId(call.request.getSrcId()),
+          nodesUtils.decodeNodeId(call.request.getSrcId())!,
```

Comment:
Similar to https://github.com/MatrixAI/js-polykey/pull/318/files#r793286473, I don't think we can make this assertion. This is on the service boundary, so we've performed no validation on the node ID to ensure it's correct.

Reply:
I must've missed that one.

Reply:
Service handlers are boundaries of the PK app on the network level. So validating the node ID by decoding it first should be done early in the service handler. If it fails to work, an exception should be thrown here. These validation exceptions can be thrown from the `src/agent/errors.ts`.

There can be a common set of exceptions that happen for both client service and agent service, and even in the CLI as well when there is some validation being done on input data before they get "decoded" into proper types. Where should these exceptions go? It's not ideal to spread them out being `src/bin/errors.ts` and `src/agent/errors.ts` and `src/client/errors.ts`. Some exceptions are put into the domains themselves `ErrorInvalidNodeId` is in `src/nodes/errors.ts` but is this actually used by the nodes domain?

For now, it can be kept in `src/nodes/errors.ts` in this situation, but in the future where there are other common exceptions, we may need to create a separate validation domain with `validation/errors.ts` to keep these errors together.

So `validateNodeId` should just be rolled into `decodeNodeId`, and try to keep the errors consistent like `ErrorXInvalid` vs `ErrorInvalidX`.

Reply:
`decodeNodeId` now validates the `NodeId` all parts of the code using this have been updated.